plugins/CoreAdminHome/Commands/InvalidateReportData.php

```diff
@@ -68,7 +71,20 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $sites = $this->getSitesToInvalidateFor($input);
         $periodTypes = $this->getPeriodTypesToInvalidateFor($input);
         $dateRanges = $this->getDateRangesToInvalidateFor($input);
-        $segments = $this->getSegmentsToInvalidateFor($input, $sites);
+
+        $segments = array();
+        if ($input->getOption('segment')) {
+            $segments = $this->getSegmentsToInvalidateFor($input, $sites);
+        }
+        if ($input->getOption('idsegments')) {
+            $idsegments = $this->getSegmentsIdToInvalidateFor($input, $sites);
+        } else {
+            $segments = $this->getSegmentsToInvalidateFor($input, $sites);
```

Comment:
@mikkeschiren wouldn't this set the segments even when not the segment option is set? like a few lines above in https://github.com/matomo-org/matomo/pull/15752/files#diff-72b51ba044329b6585537f817a184a4aR76-R77 it only executes that method if the option table is set.

I suppose it is supposed to be possible to set both `--segment=` and `--idsegments` in the same call?

I suppose it would need to look a bit like below?

```php
 $segments = $this->getSegmentsToInvalidateFor($input, $sites);
if ($input->getOption('idsegments')) {
    $idsegments = $this->getSegmentsIdToInvalidateFor($input, $sites);
    $segments = array_merge($segments, $idsegments);
    $segments = array_unique($segments);
} 
```